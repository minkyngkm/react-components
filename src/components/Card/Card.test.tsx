import { shallow } from "enzyme";
import React from "react";

import Card from "./Card";

describe("Card ", () => {
  it("renders", () => {
    const wrapper = shallow(
      <Card title="This is the title">Test content</Card>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("can display a header", () => {
    const wrapper = shallow(<Card thumbnail="test.png"></Card>);
    expect(wrapper.find(".p-card__thumbnail").prop("src")).toEqual("test.png");
  });

  it("can be highlighted", () => {
    const wrapper = shallow(<Card highlighted={true}></Card>);
    expect(wrapper.prop("className").includes("p-card--highlighted")).toBe(
      true
    );
  });

  it("can be an overlay", () => {
    const wrapper = shallow(<Card overlay={true}></Card>);
    expect(wrapper.prop("className").includes("p-card--overlay")).toBe(true);
  });

  it("can add additional classes", () => {
    const wrapper = shallow(<Card className="extra-class"></Card>);
    const className = wrapper.prop("className");
    expect(className.includes("p-card")).toBe(true);
    expect(className.includes("extra-class")).toBe(true);
  });
});
