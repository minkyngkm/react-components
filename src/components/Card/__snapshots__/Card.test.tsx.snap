// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  renders 1`] = `
<div
  className="p-card"
>
  <h3
    className="p-card__title"
  >
    This is the title
  </h3>
  <div
    className="p-card__content"
  >
    Test content
  </div>
</div>
`;
